src/sdf_tools/utils.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+
+import sdf_tools
+
+
+def compute_2d_sdf_and_gradient(grid_world, sdf_resolution, sdf_origin, frame='world'):
+    """
+    :param grid_world: a 2d numpy array of the world full of 0 and 1
+    :param sdf_resolution: float in meters
+    :param sdf_origin: size 2 np.array (vector) in meters
+    :param frame: unused
+    :return: a tuple (sdf, sdf_gradient) as numpy arrays
+    """
+    y_height = grid_world.shape[0] * sdf_resolution
+    x_width = grid_world.shape[1] * sdf_resolution
+    origin_transform = sdf_tools.Isometry3d([
+        [1.0, 0.0, 0.0, sdf_origin[0]],
+        [0.0, 1.0, 0.0, sdf_origin[1]],
+        [0.0, 0.0, 1.0, 0.0],
+        [0.0, 0.0, 0.0, 1.0]
+    ])
+
+    oob_value = sdf_tools.COLLISION_CELL(-10000)
+    occupied_value = sdf_tools.COLLISION_CELL(1)
+
+    # 2d means this should be one cell in z
+    z = sdf_resolution
+
+    grid = sdf_tools.CollisionMapGrid(origin_transform, frame, sdf_resolution, x_width, y_height, z, oob_value)
+    for x_index in range(grid.GetNumXCells()):
+        for y_index in range(grid.GetNumYCells()):
+            occupied = (grid_world[y_index, x_index] == 1)
+            if occupied:
+                grid.SetValue(x_index, y_index, 0, occupied_value)
+    sdf_result = grid.ExtractSignedDistanceField(oob_value.occupancy, False, False)
+    sdf_voxelgrid = sdf_result[0]
+    np_sdf = np.array(sdf_voxelgrid.GetRawData())
+    np_sdf = np_sdf.reshape(sdf_voxelgrid.GetNumXCells(), sdf_voxelgrid.GetNumYCells())
+    # TODO: figure out why we must transpose SDF
```

Comment:
Could these transposes be due to `(u, v)` image coordinates vs `(x,y)` coordinates you give?

Ways to check: 
1. imshow the grid world and sdf
2. numerically check points with an asymmetic grid_world